### DISCUSSION TITLE
Using CMakeTools in VSCode for Flutter Linux project, it doesnt recognize g_autoptr macro

### DISCUSSION THREAD

**Discussion by muizidn** [microsoft/vscode-cmake-tools#2940] [General]
I have a Flutter Linux project, it uses `g_autoptr` macro everywhere. Everyone can reproduce the issue by creating a new flutter project for linux platform and check main.cc in linux directory.

I open the project with VSCode and configure it with CMakeTools > Configure.

Intellisesense works fine except that it shows red squiggles for `g_autoptr`macro. When I jump to the source code, it greyed out. Even though it seems error in VSCode, it compiles successfully using `flutter build linux`

I have no experience with C/C++. Hence my assumption is that Intellisense doesn't pass the same definition as the build or something that result in `g_autoptr`unrecognized or greyed out.

Because the Intellisense uses CMakeTools configuration, then I create the question here. Any suggestion is appreciated.

![Screenshot from 2023-01-11 13-45-43](https://user-images.githubusercontent.com/19355715/211737173-24eb6223-4849-41f5-b8c7-dea5c0c57fa4.png)
![Screenshot from 2023-01-11 13-47-29](https://user-images.githubusercontent.com/19355715/211737188-1b2c93d9-c967-4bf5-a283-0fe4f4243042.png)
![Screenshot from 2023-01-11 13-48-12](https://user-images.githubusercontent.com/19355715/211737198-5bec9823-58ef-4c01-9d26-2579fbd3a311.png)

**Comment by elahehrashedi** [comment]
can you please run "CMake: log diagnostics" and share the results with us?

**Reply by bobbrow** [reply]
I tried this out.  It looks like there's a problem with cpptools not recognizing `__has_attribute(cleanup)`.  I'm asking the IntelliSense folks what might be needed to get it to return true.

**Reply by bobbrow** [reply]
It looks like there is an issue for it already: https://github.com/microsoft/vscode-cpptools/issues/10267
